### ISSUE TITLE
Persistence using Jena

### ISSUE THREAD

**Issue by simongray** [kuhumcst/DanNet#6]
Aristotle only wraps in-memory databases, but we will surely need to kind of persistence layer. Apache Jena's persistent triplestore is called TDB. There is also SDB which uses SQL databases as storage, but this has been deprecrated.

In order to use, TDB I will need to learn a bit more about the object-oriented architecture of Jena, specifically the relation between Graph vs Model vs Dataset classes.

**Comment by simongray** [comment]
* `Graph` is the underlying class.
  - Used by Aristotle.
  - Upgradable to `Model` via `Model/createModelForGraph` which returns an instance of `ModelCom`.
* Model adds the RDF layer to Graph.
  - Used by igraph-jena.
* Dataset is needed for a persistent TDB db.
  - API-wise, seems to be a fairly thin wrapper.

**Comment by simongray** [comment]
So using TDB1 as an alternative to the in-memory graph is fairly straightforward.

However, TDB2 requires wrapping every access to the graph (even regular method invocations of the various database objects) inside transactions which is very awkward and doesn't seem play well with Clojure syntax. I attempted to reorganise the code to make it work, but kept running into bugs. In the end, I decided that using TDB2 was not worth the effort and the awkward code that it produced, so TDB1 it is.
